connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresIntegrationSuite.scala

```diff
@@ -123,4 +131,33 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCT
       )
     }
   }
+
+  test("SPARK-49162: Push down filter date_trunc function") {
```

Comment:
reading the documentation for pgsql:
https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC
and databricks:
https://docs.databricks.com/en/sql/language-manual/functions/date_trunc.html

I see that they support `microseconds` while we support `microsecond`.

Can we test for all our supported precisions ? ALso can we try different casings ?

Reply:
Yes. Could we add other precisions for H2 and Postgres? @IvanK-db

Reply:
Added